### DISCUSSION TITLE
Temperature reading from another esp via network

### DISCUSSION THREAD

**Discussion by birosz** [arendst/Tasmota#11424] [Q&A] [answered]
Hello,

I just flashed the latest tasmota to a sonoff th16, works fine from the web interface. Amazing little project. I've seen it in the configuration page that it supports a few lcd displays through the i2c bus.

My question is; is there a possibility to put together another tasmota supported esp board with a display which would show the temperature measured by the sonoff? (Without altering the tasmota firmware much.)

I know if I type "http://ip/cm?cmnd=status%2010" in a browser, it will respond with:

"{"StatusSNS":{"Time":"2021-03-22T09:12:46","SI7021": "Temperature":22.3,"Humidity":48.3,"DewPoint":10.8},"TempUnit":"C"}} "

Can a tasmota flashed device do such http request to another ip? I'm not an iot programmer, but i know basic syntax like if, case, for and string operations (maybe to cut the excess text from the http reply, idk) i'm a plc programmer. Although I would not like to invest allot of time in this IF it might turn out that tasmota (or any project) would support this out of the box.

Thank you for the help in advance. I know about home assistant and mqtt and co. but I think it would be an overkill to run a server(s) just for this plus there are no other smart stuff and or control elements and scenes. The most I want to get out of this, if I succeed, to mount more sensors and maybe toggle them from the lcd with a button connected to one of the gpios.

Thanks again, any help or direction to a relevant documentation is welcome.

Have a nice day!

**Comment by barbudor** [comment, marked as answer]
If you are using MQTT this would be quite straightforward for the display unit to subscribe to the TH's SENSOR mqtt topic, extract the value and display it with some rules. 

MQTT subscribe feature is not on standard built firmware but it's easy to build your own with that feature enabled.

If your not using MQTT, your TH may directly "push" the value from a rule to the display unit with a websend command.

**Comment by birosz** [comment]
ok, thanks for the fast response, i assumed from the beginning that the mqtt can't run off an esp. i'll look into the websend rule in the tasmota documentation to see if it's simple enough for my knowledge. 
cheers

**Comment by barbudor** [comment]
The ESP are running an MQTT client but you need a MQTT server (or broker as it should be named)
If you don't want a server, http push is the way

On TH device, a rule to capture the temperature from the Télé-Sensor message and push that as an event to the display unit

On display unit, a rule triggered by the event to get the value and display it

Should be quite simple

**Comment by sfromis** [comment]
If you use MQTT, there is also a "gimmick" to let Tasmota automatically display sensor values from other Tasmota devices connected. [DisplayMode](https://tasmota.github.io/docs/Displays/#displaymode-parameters)
